cli/internal/daemon/status.go

```diff
@@ -60,22 +63,27 @@ func (l *lifecycle) status(ctx context.Context, outputJSON bool) error {
 }
 
 func (l *lifecycle) reportStatusError(err error, outputJSON bool) error {
-	var msg string
+	// Determine the unwrapped error message that we want to render.
+	var toRender error
 	if errors.Is(err, connector.ErrDaemonNotRunning) {
-		msg = "the daemon is not running"
+		toRender = connector.ErrDaemonNotRunning
+	} else if errors.Is(err, connector.ErrVersionMismatch) {
+		toRender = connector.ErrVersionMismatch
 	} else {
-		msg = err.Error()
+		toRender = err
 	}
+
```

Comment:
`err` is possibly wrapped as a `ConnectionError` at this point which means we want to reduce the amount of content we render when we're in a known error state that doesn't need so much context.

Instead of magic-stringing it like previously, we just let the error speak for itself.